Styling

samuelkinyanjui opened the pull request:
Apply React best practices and language style

samuelkinyanjui committed:
Created slice

Created two slice
books
and categories
linked the slice to the store
Added the store to application

samuelkinyanjui committed:
Fixed errors

Fixed style eerors

samuelkinyanjui committed:
Did the initial setup

Setup a Redux store
Create a slice to save your state and reducers

samuelkinyanjui committed:
updated  to add components

Use Redux in React components.

samuelkinyanjui committed:
installed react dom

installed react dom

samuelkinyanjui committed:
Created Connect API

Connect Redux with an API

samuelkinyanjui committed:
updated the read me file

updated the read me file to remove the live Demo
Update other important Information on readme file

samuelkinyanjui committed:
Updated the appication

Installed Axios
Deleted Bookslice
Deleted reducer.js
Removed comments on code

samuelkinyanjui committed:
Updated the assets

updated the asset link

samuelkinyanjui committed:
Delete .eslintrc.json

removed one .eslintrc.json files as there was error for two files

samuelkinyanjui committed:
Added styles

Added string to my app

aamirkhan2478 commented on README.md:
```diff
@@ -1,70 +1,150 @@
 
-This section has moved here: [https://facebook.github.io/create-react-app/docs/troubleshooting#npm-run-build-fails-to-minify](https://facebook.github.io/create-react-app/docs/troubleshooting#npm-run-build-fails-to-minify)
+<p align="right">(<a href="#readme-top">back to top</a>)</p>
+g
```
[OPTIONAL] Please remove this `g` letter from your README file because it make your README un professional so kindly remove it :slightly_smiling_face:

aamirkhan2478 commented on src/App.css:
- [ ] Please follow the Zaplin design for now your design is not the same as the Zaplin design if you check your design your `navbar`, `books` and `form` are not properly aligned kindly align them properly :+1: check below :point_down: the comparison between your design and Zaplin design :slightly_smiling_face: 

**Your Design**
![screencapture-localhost-3000-2023-09-15-16_14_40](https://github.com/samuelkinyanjui/bookstore_add_reducers/assets/45226848/5065ec87-4cd9-40f3-ab2c-62f5b361f011)

**Zaplin Design**
![Screenshot from 2023-09-15 16-26-18](https://github.com/samuelkinyanjui/bookstore_add_reducers/assets/45226848/06e20f4f-dfd5-4250-b3f9-bbeef4c2bdc9)

aamirkhan2478 commented on src/styles/nav.css:
- [ ] As I mentioned above please stick to the Zaplin design but your navbar is not the same as the Zaplin design if you check your navbar the bottom border is missing so kindly add it :+1:  check below :point_down: the comparison between your design and Zaplin design :slightly_smiling_face: 

**Your Design**
![Screenshot from 2023-09-15 16-26-39](https://github.com/samuelkinyanjui/bookstore_add_reducers/assets/45226848/9e453526-b3ee-4ec0-aed8-3ed50c4dbfbb)

**Zaplin Design**
![Screenshot from 2023-09-15 16-25-59](https://github.com/samuelkinyanjui/bookstore_add_reducers/assets/45226848/99ba59dd-2deb-4179-8d8f-ab9608459af1)

aamirkhan2478 commented on src/styles/nav.css:
```diff
@@ -0,0 +1,66 @@
+  margin-left: 20px;
+}
+
+nav ul li a {
+  text-decoration: none;
+}
```
- [ ] As I mentioned above please stick to the Zaplin design but your nav links are not same as Zaplin design so kindly use correct color for your nav links :+1: check below :point_down: the comparison between your design and Zaplin design :slightly_smiling_face: 

**Your Design**
![Screenshot from 2023-09-15 16-26-54](https://github.com/samuelkinyanjui/bookstore_add_reducers/assets/45226848/8dc4f3a6-89c1-4582-9044-a40a98d29324)

**Zaplin Design**
![Screenshot from 2023-09-15 16-26-30](https://github.com/samuelkinyanjui/bookstore_add_reducers/assets/45226848/0659194d-c77c-4959-9bb5-44b977d15799)

aamirkhan2478 commented on src/styles/book.css:
```diff
@@ -0,0 +1,205 @@
+  color: #595959;
+}
+
+.update-progress {
+  display: flex;
+  align-items: center;
+  justify-content: center;
+  width: 11.5rem;
+  height: 2.063rem;
+  margin: 0.75rem 0 0.563rem 3.688rem;
+  padding: 0.438rem 1.188rem 0.5rem 1.375rem;
+  border-radius: 3px;
+  background-color: #0290ff;
+}
```
- [ ] As I mentioned above please stick to the Zaplin design but your update progress button is not the same as the Zaplin design if you check your design there are black colour border on your button so kindly remove this border :+1: check below :point_down: the comparison between your design and Zaplin design :slightly_smiling_face: 

**Your Design**
![Screenshot from 2023-09-15 16-27-09](https://github.com/samuelkinyanjui/bookstore_add_reducers/assets/45226848/1be33425-04c4-4f95-9281-c06ff6c38fcc)

**Zaplin Design** 
![Screenshot from 2023-09-15 17-44-49](https://github.com/samuelkinyanjui/bookstore_add_reducers/assets/45226848/7df16c5a-4dfa-44a6-94e9-2d1c35e27cb9)

- [ ] Also if you check your font style it does not match the Zaplin design so kindly fix this issue as well :slightly_smiling_face:

aamirkhan2478 commented on src/styles/book.css:
```diff
@@ -0,0 +1,205 @@
+  border-right: 1px solid #e7e6e7;
+}
+
+.author,
+.comments,
+.edit,
+.remove button {
```
- [ ] As I mentioned above please stick to the Zaplin design but your `author`, `comments`, `edit`, and `remove` text colour and font style are not the same as the Zaplin design so kindly use the correct colour which is mentioned in the Zaplin design :+1: check below :point_down: the comparison between your design and Zaplin design :slightly_smiling_face: 

**Your Design**
- **Author Name**
![Screenshot from 2023-09-15 16-33-41](https://github.com/samuelkinyanjui/bookstore_add_reducers/assets/45226848/fb15ef75-b2c6-49b7-9a6d-70d873f09177)

- **Links** 
![Screenshot from 2023-09-15 16-28-48](https://github.com/samuelkinyanjui/bookstore_add_reducers/assets/45226848/465bb3ea-25c1-4f7c-a50e-d02d5cf1eddd)

**Zaplin Design**

- **Author Name**
![Screenshot from 2023-09-15 16-33-53](https://github.com/samuelkinyanjui/bookstore_add_reducers/assets/45226848/9eee9697-f64b-4384-8809-ce89e0f551a0)

- **Links** 
![Screenshot from 2023-09-15 16-31-51](https://github.com/samuelkinyanjui/bookstore_add_reducers/assets/45226848/734baccb-2287-419d-9647-c917f3f602c0)

aamirkhan2478 commented on src/components/Book.js:
```diff
@@ -0,0 +1,68 @@
+        <img src={progress} alt="progress" />
+        <div className="progress-info">
+          <p className="parcent">64%</p>
+          <p className="completed">Completed</p>
+        </div>
```
- [ ] As I mentioned above please stick to the Zaplin design but your Completed text colour and font style do not match the Zaplin design so kindly follow the Zaplin design as much as possible :+1: check below :point_down: the comparison between your design and Zaplin design :slightly_smiling_face: 

**Your Design**
![Screenshot from 2023-09-15 17-57-50](https://github.com/samuelkinyanjui/bookstore_add_reducers/assets/45226848/bc19bbe5-0dfc-41e9-b8d8-fa86c4dad572)

**Zaplin Design**
![Screenshot from 2023-09-15 16-34-01](https://github.com/samuelkinyanjui/bookstore_add_reducers/assets/45226848/974fc031-edee-4e60-9bac-497031696c76)

aamirkhan2478 commented on src/components/Book.js:
```diff
@@ -0,0 +1,68 @@
+          <p className="completed">Completed</p>
+        </div>
+      </div>
+      <div className="chapter">
+        <p className="current">CURRENT CHAPTER</p>
+        <p className="chapter-number">Chapter 17</p>
+        <button type="button" className="update-progress"><p>UPDATE PROGRESS</p></button>
+      </div>
```
- [ ] As I mentioned above please stick to the Zaplin design but you missed the horizontal divider between the progress bar and the Current Chapter so kindly add a horizontal divider :+1: check below :point_down: the comparison between your design and Zaplin design :slightly_smiling_face: 

**Your Design**
![Screenshot from 2023-09-15 18-04-13](https://github.com/samuelkinyanjui/bookstore_add_reducers/assets/45226848/42e57412-feae-46ea-be02-b9d090b4f2dc)

**Zaplin Design**
![Screenshot from 2023-09-15 18-04-04](https://github.com/samuelkinyanjui/bookstore_add_reducers/assets/45226848/3dccbcc9-ed16-4448-9e03-0fe203d912c7)

- [ ] Also if you check your Current Chapter text it does not match with Zaplin design so kindly fix the colour and font style as well :slightly_smiling_face:

aamirkhan2478 reviewed (changes requested):
Hi @samuelkinyanjui, 

![](https://media2.giphy.com/media/t8YPf3ACKeouLzEIvr/giphy.gif?cid=ecf05e47k5uapdptg00nidbw3qaxx8iqgsusgaswci2c3uh1&rid=giphy.gif&ct=g)

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!


## Highlights
✔️ Good Effort
✔️ Your design is good.

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

samuelkinyanjui committed:
update read me

Update read me

samuelkinyanjui committed:
Updated the files

Removed  g letter from my README
Redesigned  my navbar, books and form
I added navbar the bottom border
Used correct color
redesigned the author, comments, edit, and remove text
Added the horizontal divider between the progress bar and the Current Chapter

samuelkinyanjui committed:
updated files

Removed  g letter from my README
Redesigned  my navbar, books and form
I added navbar the bottom border
Used correct color
redesigned the author, comments, edit, and remove text
Added the horizontal divider between the progress bar and the Current Chapter

samuelkinyanjui committed:
updated files

fixed errors

Efiamotu-1 reviewed (approved):
Hi @samuelkinyanjui  

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

# STATUS: APPROVED 🕺🥳 

![image](https://user-images.githubusercontent.com/25227022/165483662-7b06c812-4cc0-40b9-a84c-4729d4cf10c3.gif)

## Highlights :dart: 
* Correct Gitflow :heavy_check_mark: 
* Good commit messages :heavy_check_mark: 
* All linters passing :heavy_check_mark: 
* Good PR summary :heavy_check_mark: 

## Optional suggestions
_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

## Important Notes 📝

- Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
- Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

> **Note:** Please, you can now Merge since it is Approved ✅, However if there are any conflicts please resolve them before merging.


## Thank you for your time 🙏 and call me if you need help or clarification ☎️

- This is My **GitHub**: [![@Efiamotu-1](https://img.shields.io/github/followers/Efiamotu-1?label=Habeeb-Efiamotu-Musa&style=social)](https://github.com/Efiamotu-1/)
- This is My **linkedIn**.[![Habeeb Efiamotu Musa](https://img.shields.io/badge/linkedin-@Habeeb-crimson)](https://www.linkedin.com/in/musa-habeeb/)
- This is My **Twitter**: [![@EFYAMOTU](https://img.shields.io/twitter/follow/omarramoun?label=Habeeb-Efiamotu-Musa&style=social)](https://twitter.com/EFYAMOTU)
- This is My **Slack**: [![Habeeb Efiamotu Musa ](https://img.shields.io/badge/slack-@Habeeb-brightgreen)](https://slack.com)

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._
